### ISSUE TITLE
Compilation of python3-timbl fails inside of Docker

### ISSUE THREAD

**Issue by jbaiter** [proycon/LaMachine#14]
I just ran `lamachine-update.sh` inside of the `proycon/lamachine` container. The process failed when trying to install `python3-timbl`:

```
Searching for python3-timbl
Reading https://pypi.python.org/simple/python3-timbl/
Best match: python3-timbl 2016.3.13
Downloading https://pypi.python.org/packages/cb/51/6ca37d2098d75f6c4f76e8355cffc38c4ef1828974006ae69fad8b80d433/python3-timbl-2016.03.13.tar.gz#md5=834cb3a035d472e9aad64119da525cb8
Processing python3-timbl-2016.03.13.tar.gz
Writing /tmp/easy_install-0zu3ilpp/python3-timbl-2016.03.13/setup.cfg
Running python3-timbl-2016.03.13/setup.py -q bdist_egg --dist-dir /tmp/easy_install-0zu3ilpp/python3-timbl-2016.03.13/egg-dist-tmp-riy_lvts
/usr/lib/python3.5/site-packages/setuptools/dist.py:285: UserWarning: Normalizing '2016.03.13' to '2016.3.13'
  normalized_version,
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
In file included from /usr/include/boost/type_traits/ice.hpp:15:0,
                 from /usr/include/boost/python/detail/def_helper.hpp:9,
                 from /usr/include/boost/python/class.hpp:29,
                 from /usr/include/boost/python.hpp:18,
                 from src/timblapi.h:53,
                 from src/timblapi.cc:47:
/usr/include/boost/type_traits/detail/ice_or.hpp:17:71: note: #pragma message: NOTE: Use of this header (ice_or.hpp) is deprecated
 # pragma message("NOTE: Use of this header (ice_or.hpp) is deprecated")
                                                                       ^
In file included from /usr/include/boost/type_traits/ice.hpp:16:0,
                 from /usr/include/boost/python/detail/def_helper.hpp:9,
                 from /usr/include/boost/python/class.hpp:29,
                 from /usr/include/boost/python.hpp:18,
                 from src/timblapi.h:53,
                 from src/timblapi.cc:47:
/usr/include/boost/type_traits/detail/ice_and.hpp:18:72: note: #pragma message: NOTE: Use of this header (ice_and.hpp) is deprecated
 # pragma message("NOTE: Use of this header (ice_and.hpp) is deprecated")
                                                                        ^
In file included from /usr/include/boost/type_traits/ice.hpp:17:0,
                 from /usr/include/boost/python/detail/def_helper.hpp:9,
                 from /usr/include/boost/python/class.hpp:29,
                 from /usr/include/boost/python.hpp:18,
                 from src/timblapi.h:53,
                 from src/timblapi.cc:47:
/usr/include/boost/type_traits/detail/ice_not.hpp:17:72: note: #pragma message: NOTE: Use of this header (ice_not.hpp) is deprecated
 # pragma message("NOTE: Use of this header (ice_not.hpp) is deprecated")
                                                                        ^
In file included from /usr/include/boost/type_traits/ice.hpp:18:0,
                 from /usr/include/boost/python/detail/def_helper.hpp:9,
                 from /usr/include/boost/python/class.hpp:29,
                 from /usr/include/boost/python.hpp:18,
                 from src/timblapi.h:53,
                 from src/timblapi.cc:47:
/usr/include/boost/type_traits/detail/ice_eq.hpp:17:71: note: #pragma message: NOTE: Use of this header (ice_eq.hpp) is deprecated
 # pragma message("NOTE: Use of this header (ice_eq.hpp) is deprecated")
                                                                       ^
src/timblapi.cc: In member function ‘boost::python::tuple TimblApiWrapper::classify(const string&)’:
src/timblapi.cc:72:19: error: could not convert ‘std::make_tuple(_Elements&& ...) [with _Elements = {bool&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&}](cls)’ from ‘std::tuple<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >’ to ‘boost::python::tuple’
  return make_tuple(result, cls);
         ~~~~~~~~~~^~~~~~~~~~~~~
src/timblapi.cc: In member function ‘boost::python::tuple TimblApiWrapper::classify2(const string&)’:
src/timblapi.cc:81:19: error: could not convert ‘std::make_tuple(_Elements&& ...) [with _Elements = {bool&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, double&}](cls, distance)’ from ‘std::tuple<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double>’ to ‘boost::python::tuple’
  return make_tuple(result, cls, distance);
         ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
src/timblapi.cc: In member function ‘boost::python::tuple TimblApiWrapper::classify3(const string&, bool, unsigned char)’:
src/timblapi.cc:96:30: error: could not convert ‘std::make_tuple(_Elements&& ...) [with _Elements = {bool, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, boost::python::dict, double&}](cls, TimblApiWrapper::dist2dict(const Timbl::ValueDistribution*, bool, double) const(distrib, ((int)normalize), (double)0), distance)’ from ‘std::tuple<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, boost::python::dict, double>’ to ‘boost::python::tuple’
             return make_tuple(true, cls, dist2dict(distrib, normalize), distance);
                    ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/timblapi.cc: In member function ‘boost::python::tuple TimblApiWrapper::classify3safe(const string&, bool, unsigned char)’:
src/timblapi.cc:158:30: error: could not convert ‘std::make_tuple(_Elements&& ...) [with _Elements = {bool, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, boost::python::dict, double&}](cls, TimblApiWrapper::dist2dict(const Timbl::ValueDistribution*, bool, double) const(distrib, ((int)normalize), (double)0), distance)’ from ‘std::tuple<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, boost::python::dict, double>’ to ‘boost::python::tuple’
             return make_tuple(true, cls, dist2dict(distrib, normalize), distance);
                    ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: Setup script exited with error: command 'gcc' failed with exit status 1
================= ERROR ===================
setup.py install gecco failed
```

**Comment by proycon** [comment]
Confirmed, this problem now reproduces locally in virtual env setting as well. I wonder when this got introduced. I'll look into fixing it.

**Comment by proycon** [comment]
This issue should now be fixed with the newly released python-timbl v2016.06.02 . It may take up to an hour for the docker image to be rebuilt still though.
